backend/ethereum/channel/funder.go

```diff
@@ -210,76 +210,61 @@ func (f *Funder) deposit(ctx context.Context, bal *big.Int, asset Asset, funding
 }
 
 // checkFunded returns whether `fundingID` holds at least `amount` funds.
-func checkFunded(ctx context.Context, amount *big.Int, asset assetHolder, fundingID [32]byte) (bool, error) {
-	iter, err := filterFunds(ctx, asset, fundingID)
+func (f *Funder) checkFunded(ctx context.Context, amount *big.Int, asset assetHolder, fundingID [32]byte) (bool, error) {
+	deposited := make(chan *subscription.Event, 10)
+	subErr := make(chan error, 1)
+	// Subscribe to events.
+	sub, err := f.depositedSub(ctx, asset.contract, fundingID)
 	if err != nil {
-		return false, errors.WithMessagef(err, "filtering old Funding events for asset %d", asset.assetIndex)
+		return false, errors.WithMessage(err, "subscribing to deposited event")
 	}
-	// nolint:errcheck
-	defer iter.Close()
+	defer sub.Close()
+	// Read from the sub.
+	go func() {
+		defer close(deposited)
+		subErr <- sub.ReadPast(ctx, deposited)
+	}()
 
 	left := new(big.Int).Set(amount)
-	for iter.Next() {
-		left.Sub(left, iter.Event.Amount)
+	for _event := range deposited {
+		event := _event.Data.(*assetholder.AssetHolderDeposited)
+		left.Sub(left, event.Amount)
 	}
-	return left.Sign() != 1, errors.Wrap(iter.Error(), "iterator")
+	return left.Sign() != 1, errors.WithMessagef(<-subErr, "filtering old Funding events for asset %d", asset.assetIndex)
 }
 
-func filterFunds(ctx context.Context, asset assetHolder, fundingIDs ...[32]byte) (*assetholder.AssetHolderDepositedIterator, error) {
-	// Filter
-	filterOpts := bind.FilterOpts{
-		Start:   uint64(1),
-		End:     nil,
-		Context: ctx}
-	iter, err := asset.FilterDeposited(&filterOpts, fundingIDs)
-	if err != nil {
-		err = cherrors.CheckIsChainNotReachableError(err)
-		return nil, errors.WithMessage(err, "filtering deposited events")
+func (f *Funder) depositedSub(ctx context.Context, contract *bind.BoundContract, fundingIDs ...[32]byte) (*subscription.EventSub, error) {
+	filter := make([]interface{}, len(fundingIDs))
+	for i, fundingID := range fundingIDs {
+		filter[i] = fundingID
 	}
-
-	return iter, nil
+	event := func() *subscription.Event {
+		return &subscription.Event{
+			Name:   "Deposited",
+			Data:   new(assetholder.AssetHolderDeposited),
+			Filter: [][]interface{}{filter},
+		}
+	}
+	sub, err := subscription.NewEventSub(ctx, f, contract, event, startBlockOffset)
```

Comment:
There is a difference in functionality here compared to before. Before, we were starting from block 1. Here we are starting from current block - offset. I think it's OK because we also do it in other places but in general this is an open issue how to choose the startBlockOffset correctly.

Reply:
I will open an issue for brainstorming it.